modules/dream2nix/WIP-python-pdm/default.nix

```diff
@@ -106,19 +123,41 @@ in {
   };
 
   public.pyEnv = let
-    pyEnv' = config.deps.python.withPackages (ps: config.mkDerivation.propagatedBuildInputs);
+    pyEnv' = config.deps.python.withPackages (
+      ps:
+        config.mkDerivation.propagatedBuildInputs
+        # the editableShellHook requires wheel and other build system deps.
+        ++ config.mkDerivation.buildInputs
+        ++ [config.deps.python.pkgs.wheel]
+    );
   in
     pyEnv'.override (old: {
       # namespaced packages are triggering a collision error, but this can be
       # safely ignored. They are still set up correctly and can be imported.
       ignoreCollisions = true;
     });
+  public.shellHook = config.pdm.editablesShellHook;
 
   public.devShell = config.deps.mkShell {
+    shellHook = config.public.shellHook;
     packages = [
       config.public.pyEnv
       config.deps.pdm
     ];
+    buildInputs =
+      [
+        config.deps.python.pkgs.tomli
```

Comment:
I think we might have to check whether our package already depends on (another version of) tomli? If so, we can just use that one and leave deps.pythonpkgs.tomli out in order to avoid conflicts.

Reply:
Oh I think I changed that bc pdm has overrides in a completely different module. I'll add that back in!

Reply:
With regards to this, should we make the pip module use the overrides module instead of pip.overrides? Then we can just put all of that logic in the new editables module, and it also makes more sense.

Reply:
Oh I think David wants the opposite: 
`overrides` and `overridesAll` to `pdm.overrides` and `pdm.overridesAll`

Reply:
I'll tentatively resolve this since I added the overrides back in.